app/src/main/java/nerd/tuxmobil/fahrplan/congress/repositories/LectureListTransformer.kt

```diff
@@ -0,0 +1,97 @@
+package nerd.tuxmobil.fahrplan.congress.repositories
+
+import android.support.v4.util.SparseArrayCompat
+import nerd.tuxmobil.fahrplan.congress.models.Lecture
+
+class LectureListTransformer(private val prioritizedRoomProvider: PrioritizedRoomProvider) {
+
+    /**
+     * Transform data loaded from the database to a data structure that we would like to use in
+     * UI code.
+     * Then transform it to the (legacy) data structure that the UI is currently using.
+     */
+    fun legacyTransformLectureList(day: Int, lectures: List<Lecture>): LegacyLectureData {
+        val dayRoomData = transformLectureList(day, lectures)
+
+        // Set Lecture.roomIndex to match the index in DayRoomData.roomDataList
+        dayRoomData.roomDataList.mapIndexed { index, roomData ->
+            roomData.lectures.forEach { it.roomIndex = index }
+        }
+
+        // Map room title to room index
+        val roomsMap = dayRoomData.roomDataList
+                .mapIndexed { index, roomData -> roomData.room to index }
+                .toMap()
+
+        // Map column index to room index
+        // This is simple identity map (0 -> 0, 1 -> 1, etc.)
+        val roomList = SparseArrayCompat<Int>()
+        for (index in 0..dayRoomData.roomDataList.size) {
+            roomList.put(index, index)
+        }
+
+        return LegacyLectureData(
+                lectureListDay = day,
+                lectureList = lectures,
+                roomsMap = roomsMap,
+                roomList = roomList,
+                roomCount = dayRoomData.roomDataList.size,
+                dayRoomData = dayRoomData
+        )
+    }
+
+    fun transformLectureList(day: Int, lectures: List<Lecture>): DayRoomData {
```

Comment:
Please rename `day` to `dayIndex`. I am trying to clean the code base from this imprecise name. Also in the other method please.